File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/domain/Events/SeatsReserved.cs
using System;
using System.Collections.Generic;
using ENode.Eventing;

namespace ConferenceManagement
{
    public class SeatsReserved : DomainEvent<Guid>
    {
        public Guid ReservationId { get; private set; }
        public IEnumerable<ReservationItem> ReservationItems { get; private set; }
        public IEnumerable<SeatAvailableQuantity> SeatAvailableQuantities { get; private set; }

        public SeatsReserved() { }
        public SeatsReserved(Guid reservationId, IEnumerable<ReservationItem> reservationItems, IEnumerable<SeatAvailableQuantity> seatAvailableQuantities)
        {
            ReservationId = reservationId;
            ReservationItems = reservationItems;
            SeatAvailableQuantities = seatAvailableQuantities;
        }
    }
}

File: samples/conference/management-bc/src/main/java/com/microsoft/conference/management/domain/events/SeatsReserved.java
package com.microsoft.conference.management.domain.events;

import com.microsoft.conference.management.domain.models.ReservationItem;
import com.microsoft.conference.management.domain.models.SeatAvailableQuantity;
import lombok.Getter;
import lombok.Setter;
import org.enodeframework.eventing.DomainEvent;

import java.util.List;

@Getter
@Setter
public class SeatsReserved extends DomainEvent<String> {
    private String reservationId;
    private List<ReservationItem> reservationItems;
    private List<SeatAvailableQuantity> seatAvailableQuantities;

    public SeatsReserved() {
    }

    public SeatsReserved(String reservationId, List<ReservationItem> reservationItems, List<SeatAvailableQuantity> seatAvailableQuantities) {
        this.reservationId = reservationId;
        this.reservationItems = reservationItems;
        this.seatAvailableQuantities = seatAvailableQuantities;
    }
}